src/transxchange/read.rs

```diff
@@ -378,90 +378,241 @@ impl Iterator for NaiveDateRange {
-                        days.insert(Fri);
-                        days.insert(Sun);
+                    "StAndrewsDayHoliday" => {
+                        dates.extend(bank_holidays.get(&SaintAndrewsHoliday).unwrap_or(&vec![]))
                     }
-                    "Weekend" => {
-                        days.insert(Sat);
-                        days.insert(Sun);
+                    "ChristmasDay" => {
+                        dates.extend(bank_holidays.get(&Christmas).unwrap_or(&vec![]))
                     }
-                    unknown_tag => warn!("Tag '{}' is not a valid tag for DaysOfWeek", unknown_tag),
-                };
+                    "ChristmasDayHoliday" => {
+                        dates.extend(bank_holidays.get(&ChristmasHoliday).unwrap_or(&vec![]))
+                    }
+                    "BoxingDay" => dates.extend(bank_holidays.get(&BoxingDay).unwrap_or(&vec![])),
+                    "BoxingDayHoliday" => {
+                        dates.extend(bank_holidays.get(&BoxingDayHoliday).unwrap_or(&vec![]))
+                    }
+                    unknown_tag => warn!(
+                        "Tag '{}' is not a valid tag BankHolidayOperation",
+                        unknown_tag
+                    ),
+                }
             }
+            dates
         }
+        // So far, `bank_holiday` only contains the Bank Holidays (no-work days)
+        // but, New Year and New Year Bank Holiday are 2 different things:
+        // - New Year is always on 1st of January
+        // - New Year Bank Holiday might be on the 2nd or 3rd if the 1st ends
+        // up being on a weekend
+        //
+        // TransXChange has a notion for both New Year and New Year Bank Holiday
+        // so we also need to add the New Year days in `bank_holidays`.
+        // The above explanation applies to:
+        // - New Year (1st of January)
+        // - 2nd of January (specific to Scotland)
+        // - Saint-Andrews (30th of November, specific to Scotland)
+        // - Christmas (25th of December)
+        // - Boxing Day (26th of December)
+        let mut bank_holidays = bank_holidays.clone();
+        let new_year_days = bank_holidays::get_fixed_days(1, 1, validity_period);
+        bank_holidays.insert(BankHoliday::NewYear, new_year_days);
+        let january_second_days = bank_holidays::get_fixed_days(2, 1, validity_period);
+        bank_holidays.insert(BankHoliday::JanuarySecond, january_second_days);
+        let saint_andrews_days = bank_holidays::get_fixed_days(30, 11, validity_period);
+        bank_holidays.insert(BankHoliday::SaintAndrews, saint_andrews_days);
+        let christmas_days = bank_holidays::get_fixed_days(25, 12, validity_period);
+        bank_holidays.insert(BankHoliday::Christmas, christmas_days);
+        let boxing_days = bank_holidays::get_fixed_days(26, 12, validity_period);
+        bank_holidays.insert(BankHoliday::BoxingDay, boxing_days);
+        let days_included =
```

Comment:
```rust
let days_included = bank_holiday_operation
            .try_only_child("DaysOfOperation")
            .map(|operation| get_bank_holidays_dates(operation, &bank_holidays))
            .unwrap_or_default();
```

Reply:
Yes, thank you, I tried to apply that everywhere I saw it.